boris_tests/BBQ_test.go

```diff
@@ -0,0 +1,110 @@
+		amount, err := Instance.CalculatePrice(42)
+		assert.Error(t, err)
+		assert.Equal(t, 0, amount)
+	})
+	t.Run("WhenTooFewEaters", func(t *testing.T) {
+		defer setup(t)()
+		amount, err := Instance.CalculatePrice(-42)
+		assert.Error(t, err)
+		assert.Equal(t, 0, amount)
+	})
+	t.Run("WhenAllGood", func(t *testing.T) {
+		const (
+			counter     = 8
+			price       = 10
+			mangalPrice = 42
+			coalPrice   = 12
+		)
+		t.Run("WithEntrecote", func(t *testing.T) {
+			mockBeef.EXPECT().GetEntrecote(counter).Times(1).Return(counter*price, nil)
+			// The next line is not needed but it is expressive and can help to understand the flow
+			mockChicken.EXPECT().GetPullet(gomock.Any()).Times(0)
+			// Next 2 rows can be moved to a separate function because they are the same in all cases of WhenAllGood part
+			// I'd do this if it is more complex than 2 rows, now the duplication is minimal and I keep it here
+			mockMangal.EXPECT().GetMangal().Times(1).Return(mangalPrice, nil)
+			mockCoal.EXPECT().GetCoal(1).Return(coalPrice, nil)
+			amount, err := Instance.CalculatePrice(counter)
+			assert.Nil(t, err)
+			assert.Equal(t, 134, amount)
+		})
+		t.Run("WithChicken", func(t *testing.T) {
+			mockBeef.EXPECT().GetEntrecote(counter).Times(1).Return(0, errors.New("No entrecote available"))
+			mockChicken.EXPECT().GetPullet(gomock.Any()).Times(1).Return(counter*price, nil)
+			mockMangal.EXPECT().GetMangal().Times(1).Return(mangalPrice, nil)
+			mockCoal.EXPECT().GetCoal(1).Return(coalPrice, nil)
+			amount, err := Instance.CalculatePrice(counter)
+			assert.Nil(t, err)
+			assert.Equal(t, 134, amount)
+		})
+	})
+
+	t.Run("WhenAllGood", func(t *testing.T) {
+		const (
+			counter     = 8
+			price       = 10
+			mangalPrice = 42
+			coalPrice   = 12
+		)
+		t.Run("WithEntrecote", func(t *testing.T) {
+			mockBeef.EXPECT().GetEntrecote(counter).Times(1).Return(counter*price, nil)
+			// The next line is not needed but it is expressive and can help to understand the flow
+			mockChicken.EXPECT().GetPullet(gomock.Any()).Times(0)
+			// Next 2 rows can be moved to a separate function because they are the same in all cases of WhenAllGood part
+			// I'd do this if it is more complex than 2 rows, now the duplication is minimal and I keep it here
+			mockMangal.EXPECT().GetMangal().Times(1).Return(mangalPrice, nil)
+			mockCoal.EXPECT().GetCoal(1).Return(coalPrice, nil)
+			amount, err := Instance.CalculatePrice(counter)
+			assert.Nil(t, err)
+			assert.Equal(t, 134, amount)
+		})
+		t.Run("WithChicken", func(t *testing.T) {
+			mockBeef.EXPECT().GetEntrecote(counter).Times(1).Return(0, errors.New("No entrecote available"))
+			mockChicken.EXPECT().GetPullet(gomock.Any()).Times(1).Return(counter*price, nil)
+			mockMangal.EXPECT().GetMangal().Times(1).Return(mangalPrice, nil)
+			mockCoal.EXPECT().GetCoal(1).Return(coalPrice, nil)
+			amount, err := Instance.CalculatePrice(counter)
+			assert.Nil(t, err)
+			assert.Equal(t, 134, amount)
+		})
+	})
```

Comment:
I didn't see test cases when GetMangal and GetCoal returns error, do they exist?

Reply:
no, they don't. There are only some examples, not a full test